pyperformance/data-files/benchmarks/bm_sqlglot/run_benchmark.py

```diff
@@ -0,0 +1,162 @@
+        "l_linenumber": "uint64",
+        "l_quantity": "float64",
+        "l_extendedprice": "float64",
+        "l_discount": "float64",
+        "l_tax": "float64",
+        "l_returnflag": "string",
+        "l_linestatus": "string",
+        "l_shipdate": "date32",
+        "l_commitdate": "date32",
+        "l_receiptdate": "date32",
+        "l_shipinstruct": "string",
+        "l_shipmode": "string",
+        "l_comment": "string",
+    },
+    "orders": {
+        "o_orderkey": "uint64",
+        "o_custkey": "uint64",
+        "o_orderstatus": "string",
+        "o_totalprice": "float64",
+        "o_orderdate": "date32",
+        "o_orderpriority": "string",
+        "o_clerk": "string",
+        "o_shippriority": "int32",
+        "o_comment": "string",
+    },
+    "customer": {
+        "c_custkey": "uint64",
+        "c_name": "string",
+        "c_address": "string",
+        "c_nationkey": "uint64",
+        "c_phone": "string",
+        "c_acctbal": "float64",
+        "c_mktsegment": "string",
+        "c_comment": "string",
+    },
+    "part": {
+        "p_partkey": "uint64",
+        "p_name": "string",
+        "p_mfgr": "string",
+        "p_brand": "string",
+        "p_type": "string",
+        "p_size": "int32",
+        "p_container": "string",
+        "p_retailprice": "float64",
+        "p_comment": "string",
+    },
+    "supplier": {
+        "s_suppkey": "uint64",
+        "s_name": "string",
+        "s_address": "string",
+        "s_nationkey": "uint64",
+        "s_phone": "string",
+        "s_acctbal": "float64",
+        "s_comment": "string",
+    },
+    "partsupp": {
+        "ps_partkey": "uint64",
+        "ps_suppkey": "uint64",
+        "ps_availqty": "int32",
+        "ps_supplycost": "float64",
+        "ps_comment": "string",
+    },
+    "nation": {
+        "n_nationkey": "uint64",
+        "n_name": "string",
+        "n_regionkey": "uint64",
+        "n_comment": "string",
+    },
+    "region": {
+        "r_regionkey": "uint64",
+        "r_name": "string",
+        "r_comment": "string",
+    },
+}
+
+
+def bench_parse():
+    parse_one(SQL)
+
+
+def bench_transpile():
+    transpile(SQL, write="spark")
+
+
+def bench_optimize():
+    optimize(parse_one(SQL), TPCH_SCHEMA)
+
+
+def bench_normalize():
+    conjunction = parse_one("(A AND B) OR (C AND D) OR (E AND F) OR (G AND H)")
+    normalize.normalize(conjunction)
+
+
+def bench_sqlglot(loops, func):
+    timer = pyperf.perf_counter()
+
+    for _ in range(loops):
+        func()
+
+    return pyperf.perf_counter() - timer
+
+
+if __name__ == "__main__":
+    runner = pyperf.Runner()
+    runner.metadata['description'] = "SQLGlot benchmark"
+    runner.bench_time_func("parse", bench_sqlglot, bench_parse)
+    runner.bench_time_func("transpile", bench_sqlglot, bench_transpile)
+    runner.bench_time_func("optimize", bench_sqlglot, bench_optimize)
+    runner.bench_time_func("normalize", bench_sqlglot, bench_normalize)
```

Comment:
Make sure to use an appropriate prefix on the names:
```suggestion
    runner.bench_time_func("sqlglot_parse", bench_sqlglot, bench_parse)
    runner.bench_time_func("sqlglot_transpile", bench_sqlglot, bench_transpile)
    runner.bench_time_func("sqlglot_optimize", bench_sqlglot, bench_optimize)
    runner.bench_time_func("sqlglot_normalize", bench_sqlglot, bench_normalize)
```